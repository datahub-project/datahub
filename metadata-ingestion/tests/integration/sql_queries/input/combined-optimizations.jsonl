{"query": "SELECT user_id, email FROM snowflake.db.users WHERE active = true", "timestamp": 1609459200, "user": "etl_job", "operation_type": "SELECT"}
{"query": "CREATE TEMPORARY TABLE temp_active_users AS SELECT user_id, email, last_login FROM snowflake.db.users WHERE active = true AND last_login > '2021-01-01'", "timestamp": 1609459260, "user": "analyst", "operation_type": "CREATE"}
{"query": "SELECT COUNT(*) as active_count FROM temp_active_users", "timestamp": 1609459320, "user": "analyst", "operation_type": "SELECT"}
{"query": "INSERT INTO snowflake.db.user_summary SELECT user_id, 'active' as status, NOW() as updated_at FROM temp_active_users", "timestamp": 1609459380, "user": "etl_job", "operation_type": "INSERT"}
{"query": "CREATE TABLE tmp_daily_metrics AS SELECT DATE(created_at) as date, COUNT(DISTINCT user_id) as unique_users FROM snowflake.db.sessions GROUP BY DATE(created_at)", "timestamp": 1609459440, "user": "analyst", "operation_type": "CREATE"}
{"query": "SELECT * FROM tmp_daily_metrics WHERE unique_users > 100 ORDER BY date DESC", "timestamp": 1609459500, "user": "analyst", "operation_type": "SELECT"}
{"query": "UPDATE snowflake.db.users SET last_seen = NOW() WHERE user_id IN (SELECT user_id FROM temp_active_users)", "timestamp": 1609459560, "user": "etl_job", "operation_type": "UPDATE"}
{"query": "DROP TABLE temp_active_users", "timestamp": 1609459620, "user": "analyst", "operation_type": "DROP"}
{"query": "DROP TABLE tmp_daily_metrics", "timestamp": 1609459680, "user": "analyst", "operation_type": "DROP"}

