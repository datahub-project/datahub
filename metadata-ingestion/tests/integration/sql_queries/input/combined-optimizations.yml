source:
  type: sql-queries
  config:
    query_file: ./input/combined-optimizations.jsonl
    platform: snowflake
    lazy_schema_resolver: true
    enable_streaming: true
    streaming_batch_size: 100
    temp_table_patterns: ["^temp_.*", "^tmp_.*"]
    use_schema_resolver: true

sink:
  type: file
  config:
    filename: ./output.json

datahub_api:
  server: http://localhost:8080
