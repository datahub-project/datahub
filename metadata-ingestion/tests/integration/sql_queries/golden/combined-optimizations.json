[
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD)",
    "changeType": "UPSERT",
    "aspectName": "upstreamLineage",
    "aspect": {
        "json": {
            "upstreams": [
                {
                    "auditStamp": {
                        "time": 1757898062510,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "created": {
                        "time": 1609459380000,
                        "actor": "urn:li:corpuser:etl_job"
                    },
                    "dataset": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)",
                    "type": "TRANSFORMED",
                    "query": "urn:li:query:composite_ded159653c305211dbfc5a30eda78ecb0669ed8deb98cbe7710c2c4f9ebedbbd"
                }
            ],
            "fineGrainedLineages": [
                {
                    "upstreamType": "FIELD_SET",
                    "upstreams": [
                        "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),user_id)"
                    ],
                    "downstreamType": "FIELD",
                    "downstreams": [
                        "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD),user_id)"
                    ],
                    "confidenceScore": 0.2,
                    "query": "urn:li:query:composite_ded159653c305211dbfc5a30eda78ecb0669ed8deb98cbe7710c2c4f9ebedbbd"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062510,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:composite_ded159653c305211dbfc5a30eda78ecb0669ed8deb98cbe7710c2c4f9ebedbbd",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "CREATE TEMPORARY TABLE temp_active_users AS SELECT user_id, email, last_login FROM snowflake.db.users WHERE active = true AND last_login > '2021-01-01';\n\nINSERT INTO snowflake.db.user_summary SELECT user_id, 'active' as status, NOW() as updated_at FROM temp_active_users;\n\nDROP TABLE temp_active_users",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459380000,
                "actor": "urn:li:corpuser:etl_job"
            },
            "lastModified": {
                "time": 1609459380000,
                "actor": "urn:li:corpuser:etl_job"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062511,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:composite_ded159653c305211dbfc5a30eda78ecb0669ed8deb98cbe7710c2c4f9ebedbbd",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)"
                },
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD),user_id)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD),status)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD),updated_at)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062511,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:composite_ded159653c305211dbfc5a30eda78ecb0669ed8deb98cbe7710c2c4f9ebedbbd",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062512,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)",
    "changeType": "UPSERT",
    "aspectName": "upstreamLineage",
    "aspect": {
        "json": {
            "upstreams": [
                {
                    "auditStamp": {
                        "time": 1757898062512,
                        "actor": "urn:li:corpuser:_ingestion"
                    },
                    "created": {
                        "time": 1609459560000,
                        "actor": "urn:li:corpuser:etl_job"
                    },
                    "dataset": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)",
                    "type": "TRANSFORMED",
                    "query": "urn:li:query:composite_f08a8699575c7cf9f02a3ff473c8d73a766c1fd23304ef7f154621d0c41ecc99"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062512,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:composite_f08a8699575c7cf9f02a3ff473c8d73a766c1fd23304ef7f154621d0c41ecc99",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "CREATE TEMPORARY TABLE temp_active_users AS SELECT user_id, email, last_login FROM snowflake.db.users WHERE active = true AND last_login > '2021-01-01';\n\nUPDATE snowflake.db.users SET last_seen = NOW() WHERE user_id IN (SELECT user_id FROM temp_active_users);\n\nDROP TABLE temp_active_users",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459560000,
                "actor": "urn:li:corpuser:etl_job"
            },
            "lastModified": {
                "time": 1609459560000,
                "actor": "urn:li:corpuser:etl_job"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062513,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:composite_f08a8699575c7cf9f02a3ff473c8d73a766c1fd23304ef7f154621d0c41ecc99",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),last_seen)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062513,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:composite_f08a8699575c7cf9f02a3ff473c8d73a766c1fd23304ef7f154621d0c41ecc99",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062514,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)",
    "changeType": "UPSERT",
    "aspectName": "datasetUsageStatistics",
    "aspect": {
        "json": {
            "timestampMillis": 1609459200000,
            "eventGranularity": {
                "unit": "DAY",
                "multiple": 1
            },
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "uniqueUserCount": 2,
            "totalSqlQueries": 3,
            "topSqlQueries": [
                "SELECT user_id, email FROM snowflake.db.users WHERE active = true",
                "CREATE TEMPORARY TABLE temp_active_users AS SELECT user_id, email, last_login FROM snowflake.db.users WHERE active = true AND last_login > '2021-01-01'",
                "UPDATE snowflake.db.users SET last_seen = NOW() WHERE user_id IN (SELECT user_id FROM temp_active_users)"
            ],
            "userCounts": [
                {
                    "user": "urn:li:corpuser:etl_job",
                    "count": 2
                },
                {
                    "user": "urn:li:corpuser:analyst",
                    "count": 1
                }
            ],
            "fieldCounts": [
                {
                    "fieldPath": "email",
                    "count": 2
                },
                {
                    "fieldPath": "user_id",
                    "count": 2
                },
                {
                    "fieldPath": "last_login",
                    "count": 1
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062514,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.sessions,PROD)",
    "changeType": "UPSERT",
    "aspectName": "datasetUsageStatistics",
    "aspect": {
        "json": {
            "timestampMillis": 1609459200000,
            "eventGranularity": {
                "unit": "DAY",
                "multiple": 1
            },
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "uniqueUserCount": 1,
            "totalSqlQueries": 1,
            "topSqlQueries": [
                "CREATE TABLE tmp_daily_metrics AS SELECT DATE(created_at) as date, COUNT(DISTINCT user_id) as unique_users FROM snowflake.db.sessions GROUP BY DATE(created_at)"
            ],
            "userCounts": [
                {
                    "user": "urn:li:corpuser:analyst",
                    "count": 1
                }
            ],
            "fieldCounts": [
                {
                    "fieldPath": "created_at",
                    "count": 1
                },
                {
                    "fieldPath": "user_id",
                    "count": 1
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062515,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD)",
    "changeType": "UPSERT",
    "aspectName": "operation",
    "aspect": {
        "json": {
            "timestampMillis": 1757898062516,
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "actor": "urn:li:corpuser:etl_job",
            "operationType": "INSERT",
            "sourceType": "DATA_PLATFORM",
            "lastUpdatedTimestamp": 1609459380000,
            "queries": [
                "urn:li:query:acb4be0ca0363397949fbdc0bb6e2e1e4d1d8642cb978a7d3666388985452741"
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062516,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:acb4be0ca0363397949fbdc0bb6e2e1e4d1d8642cb978a7d3666388985452741",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "INSERT INTO snowflake.db.user_summary SELECT user_id, 'active' as status, NOW() as updated_at FROM temp_active_users",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459380000,
                "actor": "urn:li:corpuser:etl_job"
            },
            "lastModified": {
                "time": 1609459380000,
                "actor": "urn:li:corpuser:etl_job"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062516,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:acb4be0ca0363397949fbdc0bb6e2e1e4d1d8642cb978a7d3666388985452741",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,temp_active_users,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,temp_active_users,PROD),user_id)"
                },
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD),user_id)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD),status)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.user_summary,PROD),updated_at)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062517,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:acb4be0ca0363397949fbdc0bb6e2e1e4d1d8642cb978a7d3666388985452741",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062517,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)",
    "changeType": "UPSERT",
    "aspectName": "operation",
    "aspect": {
        "json": {
            "timestampMillis": 1757898062517,
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "actor": "urn:li:corpuser:etl_job",
            "operationType": "UPDATE",
            "sourceType": "DATA_PLATFORM",
            "lastUpdatedTimestamp": 1609459560000,
            "queries": [
                "urn:li:query:0d70cbb6e646505408896fb071c533336d1bd7c3c6544535b0fa406f27888b77"
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062518,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:0d70cbb6e646505408896fb071c533336d1bd7c3c6544535b0fa406f27888b77",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "UPDATE snowflake.db.users SET last_seen = NOW() WHERE user_id IN (SELECT user_id FROM temp_active_users)",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459560000,
                "actor": "urn:li:corpuser:etl_job"
            },
            "lastModified": {
                "time": 1609459560000,
                "actor": "urn:li:corpuser:etl_job"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062518,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:0d70cbb6e646505408896fb071c533336d1bd7c3c6544535b0fa406f27888b77",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)"
                },
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,temp_active_users,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),last_seen)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062519,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:0d70cbb6e646505408896fb071c533336d1bd7c3c6544535b0fa406f27888b77",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062519,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:7a22690f4c747f679b2c88fa2c206ded23f7437a4ac8519289681c498ce6632b",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "SELECT user_id, email FROM snowflake.db.users WHERE active = true",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459200000,
                "actor": "urn:li:corpuser:etl_job"
            },
            "lastModified": {
                "time": 1609459200000,
                "actor": "urn:li:corpuser:etl_job"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062519,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:7a22690f4c747f679b2c88fa2c206ded23f7437a4ac8519289681c498ce6632b",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),email)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),user_id)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062520,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:7a22690f4c747f679b2c88fa2c206ded23f7437a4ac8519289681c498ce6632b",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062520,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:7080dde05841d7e1c9850bba9a8b555163ae6bc4d37ec22cfba8f6d3e0155398",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "CREATE TEMPORARY TABLE temp_active_users AS SELECT user_id, email, last_login FROM snowflake.db.users WHERE active = true AND last_login > '2021-01-01'",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459260000,
                "actor": "urn:li:corpuser:analyst"
            },
            "lastModified": {
                "time": 1609459260000,
                "actor": "urn:li:corpuser:analyst"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062521,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:7080dde05841d7e1c9850bba9a8b555163ae6bc4d37ec22cfba8f6d3e0155398",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),email)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),last_login)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD),user_id)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062521,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:7080dde05841d7e1c9850bba9a8b555163ae6bc4d37ec22cfba8f6d3e0155398",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062521,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:8d11076530430fb59f6175aa21a788390cff300b928f5dd7bdcecf8de47fc79b",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "CREATE TABLE tmp_daily_metrics AS SELECT DATE(created_at) as date, COUNT(DISTINCT user_id) as unique_users FROM snowflake.db.sessions GROUP BY DATE(created_at)",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459440000,
                "actor": "urn:li:corpuser:analyst"
            },
            "lastModified": {
                "time": 1609459440000,
                "actor": "urn:li:corpuser:analyst"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062522,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:8d11076530430fb59f6175aa21a788390cff300b928f5dd7bdcecf8de47fc79b",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.sessions,PROD)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.sessions,PROD),created_at)"
                },
                {
                    "entity": "urn:li:schemaField:(urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.sessions,PROD),user_id)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062522,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:8d11076530430fb59f6175aa21a788390cff300b928f5dd7bdcecf8de47fc79b",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062523,
        "runId": "sql-queries-2025_09_14-21_01_02-mfvyni",
        "lastRunId": "no-run-id-provided"
    }
}
]